manager/integration/tests/test_backing_image.py

```diff
@@ -648,4 +650,46 @@ def test_volume_wait_for_backing_image_conditino(): # NOQA
       would be first True and then change to False when
       the BackingImage is ready and all the replicas are in running state.
     """
-    pass
+    # Create a large volume and export as backingimage
+    lht_host_id = get_self_host_id()
+
+    volume1_name = "vol1"
+    volume1 = create_and_check_volume(
+        client, volume1_name, 3,
+        str(2 * Gi))
+    volume1.attach(hostId=lht_host_id)
+    volume1 = wait_for_volume_healthy(client, volume1_name)
+    volume_endpoint = get_volume_endpoint(volume1)
+    write_volume_dev_random_mb_data(volume_endpoint, 1, 1000)
+    vol1_cksum = get_device_checksum(volume_endpoint)
+
+    backing_img_name = 'bi-test'
+    backing_img = client.create_backing_image(
+            name=backing_img_name,
+            sourceType=BACKING_IMAGE_SOURCE_TYPE_FROM_VOLUME,
+            parameters={"export-type": "qcow2", "volume-name": volume1_name},
+            expectedChecksum="")
+
+    # Create volume with that backing image
+    volume2_name = "vol2"
+    volume2 = create_and_check_volume(
+        client, volume_name=volume2_name, size=str(2 * Gi),
+        backing_image=backing_img["name"])
+
+    volume2.attach(hostId=lht_host_id)
+
+    check_backing_image_disk_map_status(client,
+                                        backing_img_name,
+                                        1,
+                                        BACKING_IMAGE_STATE_IN_PROGRESS)
```

Comment:
1. Does all the disk shows `in-progress` status at once or one disk shows `in-progress` and other show `pending` or `waiting`?
2. What's the behavior when backingimage status is complete on only one disk? Does volume utilizes that and become healthy?
3. Is it possible to write the `in-progress` check in if statement to make code more readable?
```
if Backingimage is in progress:
    volume2 = client.by_id_volume(volume2_name)
    assert volume2.conditions.WaitForBackingImage.status == "True"
```

Reply:
Hi @khushboo-rancher,
1. At beginning of backingimage start, only 1 diskFileStatusMap exist and the other 2 were empty. from UI can see "File processing is not started".

  <details>
    <summary>In progress backingimage yaml</summary>
  
  ```yaml
 > k get backingimage -A -o yaml
apiVersion: v1
items:
- apiVersion: longhorn.io/v1beta2
  kind: BackingImage
  metadata:
    creationTimestamp: "2024-01-05T03:55:55Z"
    finalizers:
    - longhorn.io
    generation: 4
    labels:
      longhorn.io/component: backing-image
      longhorn.io/managed-by: longhorn-manager
    name: bi-test
    namespace: longhorn-system
    resourceVersion: "45785"
    uid: d7399567-f041-404f-8b84-c671f699e621
  spec:
    checksum: ""
    disks:
      ae46ff66-a2d5-4475-b1e4-2b774b172495: ""
      b387e856-7849-4e1c-8e2c-f9e8cc54a0ee: ""
      b2386520-4a2c-4d8b-8ae1-59a3d424a034: ""
    sourceParameters:
      export-type: qcow2
      volume-name: vol1
    sourceType: export-from-volume
  status:
    checksum: ""
    diskFileStatusMap:
      b2386520-4a2c-4d8b-8ae1-59a3d424a034:
        lastStateTransitionTime: "2024-01-05T03:56:04Z"
        message: ""
        progress: 48
        state: in-progress
    diskLastRefAtMap: {}
    ownerID: ip-172-31-39-108
    size: 2147483648
    uuid: d50fa650
kind: List
metadata:
  resourceVersion: ""
  ```
  </details>

 ![image](https://github.com/longhorn/longhorn-tests/assets/73337386/cf03d137-35ea-4613-ad35-dd8bd79ec283)

2. After first 1 disk complete(in state `ready`), then the other 2 disks started( have state `in-progress`), in the meantime, volume kept in attaching state, until backingimage on 3 disks all ready, volume become healthy.
![image](https://github.com/longhorn/longhorn-tests/assets/73337386/131dd0b5-9027-46a3-88db-04618cbbb8d0)
 
3. Updated!